sdk/js/src/Client.ts

```diff
@@ -216,7 +216,7 @@ export class DevCycleClient<
 
         this.eventEmitter.emitInitialized(true)
 
-        if (initialUser.isAnonymous) {
```

Comment:
i thought it was the intention that the user passed into the SDK needed to either have isAnonymous=true, or have a user id set, otherwise you'd receive an error? That's what the comments on the DevCycleUser interface say anyway

Reply:
It looks like that's an old comment that wasn't removed when this change was made. After anonymous user ID caching was implemented, if the user ID isn't passed then it's assumed the user is anonymous https://github.com/DevCycleHQ/js-sdks/blob/3ee5528956a09afae75c921fcbc9521b15fff551/sdk/js/src/User.ts#L52

UofPx has been using the SDK with an empty user object, and as a result of this each initialization clears the ID cache and counts as an MAU

Reply:
ok. Can you fix that comment and maybe add a test for this?